### PR TITLE
Update config preferences after updating NVTs (9.0)

### PR DESCRIPTION
**Pull request by timopollmeier**
This ensures the configs still apply if preference names are changed in
the NVTs.

**Checklist**:

- Tests N/A
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/manage_sql_nvts.c [hunk 1]
```diff
@@ -1558,13 +1636,20 @@ manage_update_nvt_cache_osp (const gchar *update_socket)
                    "                 0);")
           == 0)
         {
-          check_preference_names ("config_preferences");
-          check_preference_names ("config_preferences_trash");
+          check_old_preference_names ("config_preferences");
+          check_old_preference_names ("config_preferences_trash");
+
+          /* Force update of names in new format in case hard-coded names
+           *  in migrator are outdated */
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]:
```suggestion
           * in migrator are outdated */
```

File: src/manage_sql_nvts.c [hunk 2]
```diff
@@ -1410,6 +1410,75 @@ check_preference_names (const gchar *table)
...
+static void
+check_preference_names (int trash, time_t modification_time)
+{
+  iterator_t prefs;
+
+  sql_begin_immediate ();
+
+  init_iterator (&prefs,
+                 "WITH new_pref_matches AS"
+                 " (SELECT substring(nvt_preferences.name,"
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]:
```suggestion
                 " (SELECT substring (nvt_preferences.name,"
```

File: src/manage_sql_nvts.c [hunk 3]
```diff
@@ -1410,6 +1410,75 @@ check_preference_names (const gchar *table)
...
+                 "                   '^([^:]*:[^:]*)') || ':%%' AS match,"
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]:
```suggestion
                 "                    '^([^:]*:[^:]*)') || ':%%' AS match,"
```

File: src/manage_sql_nvts.c [hunk 4]
```diff
@@ -1410,6 +1410,75 @@ check_preference_names (const gchar *table)
...
+                 "          name AS new_name"
+                 "     FROM nvt_preferences"
+                 "    WHERE substr (name, 0, position(':' IN name))"
```
Comment (mattmundell) [inline src/manage_sql_nvts.c]:
```suggestion
                 "    WHERE substr (name, 0, position (':' IN name))"
```

At least we should be consistent about this within a single statement :)

File: src/manage_sql_nvts.c (hunk 1 above)
Comment (timopollmeier) [reply src/manage_sql_nvts.c]: I've rewritten this line in 991b9d11dfa93cde1a6eaa31aec557446570e3a3.

**Pull request merged**